src/control/security/grpc_authorization_test.go

```diff
@@ -53,65 +57,152 @@ func TestCommonNameToComponent(t *testing.T) {
-		{"StartRanks", ComponentServer, "/mgmt.MgmtSvc/StartRanks"},
+func TestSecurity_ComponentHasAccess(t *testing.T) {
+	allComponents := []Component{ComponentUndefined, ComponentAdmin, ComponentAgent, ComponentServer}
+	testCases := map[string]Component{
+		"/ctl.MgmtCtl/StoragePrepare":     ComponentAdmin,
+		"/ctl.MgmtCtl/StorageScan":        ComponentAdmin,
+		"/ctl.MgmtCtl/StorageFormat":      ComponentAdmin,
+		"/ctl.MgmtCtl/SystemQuery":        ComponentAdmin,
+		"/ctl.MgmtCtl/SystemStop":         ComponentAdmin,
+		"/ctl.MgmtCtl/SystemResetFormat":  ComponentAdmin,
+		"/ctl.MgmtCtl/SystemStart":        ComponentAdmin,
+		"/ctl.MgmtCtl/NetworkScan":        ComponentAdmin,
+		"/ctl.MgmtCtl/FirmwareQuery":      ComponentAdmin,
+		"/ctl.MgmtCtl/FirmwareUpdate":     ComponentAdmin,
+		"/mgmt.MgmtSvc/Join":              ComponentServer,
+		"/mgmt.MgmtSvc/LeaderQuery":       ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolCreate":        ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolDestroy":       ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolQuery":         ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolSetProp":       ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolGetACL":        ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolOverwriteACL":  ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolUpdateACL":     ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolDeleteACL":     ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolExclude":       ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolDrain":         ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolReintegrate":   ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolEvict":         ComponentAdmin,
+		"/mgmt.MgmtSvc/PoolExtend":        ComponentAdmin,
+		"/mgmt.MgmtSvc/GetAttachInfo":     ComponentAgent,
+		"/mgmt.MgmtSvc/SmdQuery":          ComponentAdmin,
+		"/mgmt.MgmtSvc/ListPools":         ComponentAdmin,
+		"/mgmt.MgmtSvc/ListContainers":    ComponentAdmin,
+		"/mgmt.MgmtSvc/ContSetOwner":      ComponentAdmin,
+		"/mgmt.MgmtSvc/PrepShutdownRanks": ComponentServer,
+		"/mgmt.MgmtSvc/StopRanks":         ComponentServer,
+		"/mgmt.MgmtSvc/PingRanks":         ComponentServer,
+		"/mgmt.MgmtSvc/ResetFormatRanks":  ComponentServer,
+		"/mgmt.MgmtSvc/StartRanks":        ComponentServer,
 	}
 
-	if len(testCases) != len(methodAuthorizations) {
-		t.Errorf("component access tests is missing a case for a newly added method")
+	var missing []string
+	for method := range methodAuthorizations {
+		if _, found := testCases[method]; !found {
+			missing = append(missing, method)
+		}
 	}
-	for _, tc := range testCases {
-		t.Run(tc.testname, func(t *testing.T) {
+	if len(missing) > 0 {
+		t.Fatalf("%s is missing test cases for the following methods:\n%s", t.Name(), strings.Join(missing, "\n"))
+	}
+
+	var invalid []string
+	for testMethod := range testCases {
+		if _, found := methodAuthorizations[testMethod]; !found {
+			invalid = append(invalid, testMethod)
+		}
+	}
+	if len(invalid) > 0 {
+		t.Fatalf("%s has test cases for the following invalid methods:\n%s", t.Name(), strings.Join(invalid, "\n"))
```

Comment:
Might be better to phrase this as "tests exist for methods not in the authorization table." I got a little confused at first thinking the tests below were dupes of these checks. The methods here aren't necessarily invalid, just not in the auth table yet.

Reply:
It's more likely that the tests were added for methods which were later removed (e.g. BioHealthQuery, etc). Mostly just trying to avoid buildup of cruft.

Reply:
I've updated the message to make it clearer that the problem is that the tested method isn't in the authorization map.